### PR TITLE
Modified LeMat Fluff Weapon

### PR DESCRIPTION
**Pull request by HzDonut**
**Approval:** https://forum.vore-station.net/viewtopic.php?f=27&t=1243

**Ckey and Character Name:** HzDonut - Jesse Soemmer

**PR Includes:** Modified LeMat (cannot fire shotgun shells or load secondary barrel), Sidearm permit.

**Additional Information:** Would appreciate a holster to go with this since carrying a self defense weapon around in your backpack seems a bit odd, dontcha think?

### CODE REVIEW

**Comment by SpadesNeil** [comment]
I mean you could do it this way but it's a clunky and badly coded way of doing it.

It'll work, but I won't like it. :U

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 1]
```diff
@@ -347,6 +347,18 @@
 	name = "\improper MEUSOC .45"
 	desc = "Some serious drywall work, coming up!"
 */
+//hzdonut:Jesse Soemmer
+/obj/item/weapon/gun/projectile/revolver/fluff/jesselemat
+	name = "LeMat Revolver"
+	desc = "The LeMat Revolver is a 9 shot revolver with a secondary firing barrel for loading shotgun shells. For when you really need something dead. Uses .38-Special and 12g rounds depending on the barrel. This one appears to have had it's secondary barrel sealed off, however, and looks to be in pristine condition. Either it's brand new, or its owner takes very good care of it."
+	icon_state = "lemat"
+	item_state = "revolver"
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: This line is redundant. Inherited from /revolver

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 2]
```diff
@@ -347,6 +347,18 @@
...
+	caliber = ".38"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
+	handle_casings = CYCLE_CASINGS
+	max_shells = 9
+	ammo_type = /obj/item/ammo_casing/a38
+	projectile_type = /obj/item/projectile/bullet/pistol
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: This line shouldn't exist. Delete this.

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 3]
```diff
@@ -347,6 +347,18 @@
...
 */
+//hzdonut:Jesse Soemmer
+/obj/item/weapon/gun/projectile/revolver/fluff/jesselemat
+	name = "LeMat Revolver"
+	desc = "The LeMat Revolver is a 9 shot revolver with a secondary firing barrel for loading shotgun shells. For when you really need something dead. Uses .38-Special and 12g rounds depending on the barrel. This one appears to have had it's secondary barrel sealed off, however, and looks to be in pristine condition. Either it's brand new, or its owner takes very good care of it."
+	icon_state = "lemat"
+	item_state = "revolver"
+	caliber = ".38"
+	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
+	handle_casings = CYCLE_CASINGS
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: This line is redundant. Inherited from /revolver

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 4]
```diff
@@ -347,6 +347,18 @@
 	name = "\improper MEUSOC .45"
 	desc = "Some serious drywall work, coming up!"
 */
+//hzdonut:Jesse Soemmer
+/obj/item/weapon/gun/projectile/revolver/fluff/jesselemat
+	name = "LeMat Revolver"
+	desc = "The LeMat Revolver is a 9 shot revolver with a secondary firing barrel for loading shotgun shells. For when you really need something dead. Uses .38-Special and 12g rounds depending on the barrel. This one appears to have had it's secondary barrel sealed off, however, and looks to be in pristine condition. Either it's brand new, or its owner takes very good care of it."
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: This is crudely copypasted.

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 5]
```diff
@@ -347,6 +347,18 @@
 	name = "\improper MEUSOC .45"
 	desc = "Some serious drywall work, coming up!"
 */
+//hzdonut:Jesse Soemmer
+/obj/item/weapon/gun/projectile/revolver/fluff/jesselemat
+	name = "LeMat Revolver"
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: This is crudely copypasted.

**Comment by SpadesNeil** [comment]
An infinitely better way of coding this would be just to edit only what you're changing in the lemat revolver.

Do what you did before. lemat/fluff/youritem, then only add variables for stuff that's different from the original.

Do that, and I'll do the part about disabling the weapon selection mode.

**Comment by Screemonster** [comment]
Personally, rather than making it a revolver/lemat and then disabling all the things that make a lemat a lemat, I'd rather just make it a revolver and just not enable those things.

**Comment by HzDonut** [comment]
The default revolver in the code is .357 with the one labelled 'detective revolver' being the .38. I just copied the revolver code, made it .38, and renamed it to LeMat with the LeMat icon. Why's this a bad fix exactly? It does exactly what you so asked me to do.

**Comment by HzDonut** [comment]
There, I changed the "crude copypasting" and deleted the things that were actually the same as the information on the revolver code.

**Comment by HzDonut** [comment]
The revolver's code is this:
`/obj/item/weapon/gun/projectile/revolver
	name = "revolver"
	desc = "The Lumoco Arms HE Colt is a choice revolver for when you absolutely, positively need to put a hole in the other guy. Uses .357 rounds."
	icon_state = "revolver"
	item_state = "revolver"
	caliber = ".357"
	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
	handle_casings = CYCLE_CASINGS
	max_shells = 6
	ammo_type = /obj/item/ammo_casing/a357
	projectile_type = /obj/item/projectile/bullet/pistol/strong
	var/chamber_offset = 0 //how many empty chambers in the cylinder until you hit a **round`**

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 6]
```diff
@@ -347,6 +347,15 @@
 	name = "\improper MEUSOC .45"
 	desc = "Some serious drywall work, coming up!"
 */
+//hzdonut:Jesse Soemmer
+/obj/item/weapon/gun/projectile/revolver/fluff/jesselemat
+	name = "Modified LeMat"
+	desc = "The LeMat Revolver is a 9 shot revolver with a secondary firing barrel for loading shotgun shells. Uses .38-Special and 12g rounds depending on the barrel. This one appears to have had it's secondary barrel sealed off and looks to be in pristine condition. Either it's brand new, or its owner takes very good care of it."
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: Now this is much nicer.

File: code/modules/vore/fluffstuff/custom_guns_vr.dm [hunk 7]
```diff
@@ -347,6 +347,15 @@
...
+	icon_state = "lemat"
+	caliber = ".38"
+	max_shells = 9
+	ammo_type = /obj/item/ammo_casing/a38
+	projectile_type = /obj/item/projectile/bullet/pistol
```
Comment (SpadesNeil) [inline code/modules/vore/fluffstuff/custom_guns_vr.dm]: This line still needs to be deleted. It's dead code.

Comment (HzDonut) [reply code/modules/vore/fluffstuff/custom_guns_vr.dm]: If you look at the revolver code, you'll see that it's all .357, the revolver with .38 ammo is the detective revolver that has it's own set of special nonsense. If you're saying that deleting this won't cause me to end up with a revolver that shoots .357 ammo though i'll believe ya.

**Comment by HzDonut** [comment]
That should've fixed all the problems brought up, I hope.

**Comment by HzDonut** [comment]
It's been eight days and the problem has been fixed this entire time, why is this still on do not merge?

File: code/modules/vore/fluffstuff/custom_guns_vr.dm (hunk 7 above)
Comment (SpadesNeil) [reply code/modules/vore/fluffstuff/custom_guns_vr.dm]: `projectile_type = /obj/item/projectile/bullet/pistol` doesn't even belong in the .357 code. I should fix that.

`ammo_type = /obj/item/ammo_casing/a38` is necessary to decide what ammo spawns in the gun to start with.

Comment (HzDonut) [reply code/modules/vore/fluffstuff/custom_guns_vr.dm]: So I should re-add the ammo_type = /obj/item/ammo_casing/a38 code then? If I delete the line about /obj/item/projectile/bullet/pistol then it _should_ instead take the /obj/item/projectile/bullet/pistol/strong code from the revolver code which is equivalent to the Mateba, which I'm sure you guys don't want.

Comment (SpadesNeil) [reply code/modules/vore/fluffstuff/custom_guns_vr.dm]: That should not be in the original code, nor does it actually do anything to begin with. It should make no difference if it's removed and I should go remove it from Polaris.

Comment (HzDonut) [reply code/modules/vore/fluffstuff/custom_guns_vr.dm]: Ah, I see. Alright then, I'll leave things untouched.

**Pull request merged**